<div class="alert alert-block alert-info">
<center>ESTATÍSTICA APLICADA</center>
<center>PROJETO 01</center>
<center>ALUNO:  Daniel Barbosa Vasconcelos</center>
</div>

In [39]:
import numpy as np
import pandas as pd
import zipfile

<div class="alert alert-block alert-info">
    1° Tarefa
<div>

In [41]:
# Pegando os dados em zip
dados_prf_2018 = pd.read_csv('https://arquivos.prf.gov.br/arquivos/index.php/s/MaC6cieXSFACNWT/download',
                        encoding = 'latin1', compression = 'zip', sep = ';')
dados_prf_2019 = pd.read_csv('https://arquivos.prf.gov.br/arquivos/index.php/s/kRBUylqz6DyQznN/download',
                        encoding = 'latin1', compression = 'zip', sep = ';')
dados_prf_2020 = pd.read_csv('https://arquivos.prf.gov.br/arquivos/index.php/s/jdDLrQIf33xXSCe/download',
                        encoding = 'latin1', compression = 'zip', sep = ';')
dados_prf_2021 = pd.read_csv('http://arquivos.prf.gov.br/arquivos/index.php/s/n1T3lymvIdDOzzb/download',
                        encoding = 'latin1', compression = 'zip', sep = ';')

# juntando todo os dataframes com concat
juntando_dados = pd.concat([dados_prf_2018,dados_prf_2019 ,dados_prf_2020,dados_prf_2021],
                          keys = ['2018', '2019', '2020', '2021'])
juntando_dados

id data_inversa     dia_semana   horario  uf     br     km  \
2018 0       99973.0   2018-01-01  segunda-feira  00:20:00  RJ  116.0  303,5   
     1       99976.0   2018-01-01  segunda-feira  00:40:00  SC  282.0    0,4   
     2       99977.0   2018-01-01  segunda-feira  00:30:00  RJ  493.0      1   
     3       99981.0   2018-01-01  segunda-feira  01:15:00  RS  386.0    134   
     4       99982.0   2018-01-01  segunda-feira  00:20:00  RS  293.0  151,7   
...              ...          ...            ...       ...  ..    ...    ...   
2021 64510  425426.0   2021-05-13   quinta-feira  03:15:00  MG  381.0  775,3   
     64511  425436.0   2021-11-15  segunda-feira  23:00:00  PA   10.0  248,4   
     64512  428107.0   2021-09-21    terça-feira  01:10:00  MG   50.0     20   
     64513  435016.0   2021-12-29   quarta-feira  13:40:00  AC  364.0    170   
     64514  441173.0   2021-08-22        domingo  17:10:00  MT  364.0    498   

                 municipio                                     causa_acidente  \
2018 0             RESENDE                                  Condutor Dormindo   
     1       FLORIANOPOLIS                 Não guardar distância de segurança   
     2            ITABORAI                             Ultrapassagem Indevida   
     3             SARANDI                                 Ingestão de Álcool   
     4            CANDIOTA                        Falta de Atenção à Condução   
...                    ...                                                ...   
2021 64510        CAMPANHA  Condutor deixou de manter distância do veículo...   
     64511  AURORA DO PARA                             Transitar na contramão   
     64512        ARAGUARI                                  Condutor Dormindo   
     64513          BUJARI                                   Pista esburacada   
     64514         JANGADA           Reação tardia ou ineficiente do condutor   

                           tipo_acidente  ... feridos_graves ilesos ignorados  \
2018 0         Saída de leito carroçável  ...              4      1         0   
     1                  Colisão traseira  ...              2      1         0   
     2                   Colisão frontal  ...              0      3         1   
     3               Colisão transversal  ...              0      2         0   
     4         Saída de leito carroçável  ...              1      0         0   
...                                  ...  ...            ...    ...       ...   
2021 64510              Colisão traseira  ...              0      2         0   
     64511           Colisão transversal  ...              1      1         0   
     64512            Colisão com objeto  ...              2      1         0   
     64513  Queda de ocupante de veículo  ...              1      0         0   
     64514              Colisão traseira  ...              0      1         0   

           feridos veiculos      latitude     longitude  regional  delegacia  \
2018 0           4        1     -22,46937     -44,44705     SR-RJ     DEL5/7   
     1           2        2  -27,59971687  -48,57565694     SR-SC     DEL8/1   
     2           1        3    -22,763901    -42,927532     SR-RJ     DEL5/4   
     3           0        2    -27,953636    -52,916374     SR-RS    DEL9/14   
     4           1        1    -31,395214  -53,78391174     SR-RS    DEL9/11   
...            ...      ...           ...           ...       ...        ...   
2021 64510       2        2    -21.790183    -45.446659     SR-MG      UOP02   
     64511       2        2     -2.267754    -47.563084     SR-PA      UOP01   
     64512       7        1     -18.58721    -48.056456     SR-MG      UOP02   
     64513       1        1     -9.766854    -68.056811     SR-AC      UOP01   
     64514       1        2    -15.263832    -56.455487     SR-MT      UOP02   

                 uop  
2018 0      UOP03/RJ  
     1      UOP01/SC  
     2      UOP02/RJ  
     3      UOP01/RS  
     4      UOP03/RS  
...         

<div class="alert alert-block alert-info">
    2° Tarefa
<div>

<div class="alert alert-block alert-info" style="background-color:#b9ceac; color:#000; border-radius:15px;" >
    a)
<div>

In [42]:
# quantos acidentes ocorreram em cada ano.

print(' Em 2018 ocorreu', 
      len(dados_prf_2018), '\n Em 2019 ocorreu',
      len(dados_prf_2019), '\n Em 2020 ocorreu',
      len(dados_prf_2020), '\n em 2021 ocorreu',
     len(dados_prf_2021))

 Em 2018 ocorreu 69295 
 Em 2019 ocorreu 67446 
 Em 2020 ocorreu 63548 
 em 2021 ocorreu 64515


<div class="alert alert-block alert-info" style="background-color:#b9ceac; color:#000; border-radius:15px;" >
    b)
<div>

In [43]:
# quantas variáveis cada conjunto de dados registra?

print('2018 teve registada de variáveis', 
      len(dados_prf_2018.columns), '\n2019 teve registada de variáveis',
      len(dados_prf_2019.columns), '\n2020 teve registada de variáveis',
      len(dados_prf_2020.columns), '\n2021 teve registada de variáveis',
     len(dados_prf_2021.columns))

2018 teve registada de variáveis 30 
2019 teve registada de variáveis 30 
2020 teve registada de variáveis 30 
2021 teve registada de variáveis 30


<div class="alert alert-block alert-info" style="background-color:#b9ceac; color:#000; border-radius:15px;" >
    c)
<div>

In [44]:
# Quais cidades tiveram mais acidentes nas rodovias federaisem em três anos.

todos_os_dados = juntando_dados.groupby(['municipio']).size().reset_index(name = 'count')
todos_os_dados = todos_os_dados.sort_values(by = 'count', ascending=False)
todos_os_dados.head()

municipio  count
253    BRASILIA   3978
551    CURITIBA   3962
1680   SAO JOSE   3011
743   GUARULHOS   2781
1246    PALHOCA   2496

<div class="alert alert-block alert-info" style="background-color:#b9ceac; color:#000; border-radius:15px;" >
    d)
<div>

In [45]:
# Quantidades de acidentes em todos os anos separadamente e juntos nos três anos

acidentes_pb_2018 = dados_prf_2018[(dados_prf_2018.uf == 'PB') & (dados_prf_2018.feridos_graves >= 1)]
acidentes_pb_2019 = dados_prf_2019[(dados_prf_2019.uf == 'PB') & (dados_prf_2019.feridos_graves >= 1)]
acidentes_pb_2020 = dados_prf_2020[(dados_prf_2020.uf == 'PB') & (dados_prf_2020.feridos_graves >= 1)] 
acidentes_pb_2021 = dados_prf_2021[(dados_prf_2021.uf == 'PB') & (dados_prf_2021.feridos_graves >= 1)]
acidentes_pb_3anos = juntando_dados[(juntando_dados.uf == 'PB') & (juntando_dados.feridos_graves >= 1)]

print('A Paraiba teve', len(acidentes_pb_2018), ' de acidentes com pessoas gravemente feridas em 2018')
print('A Paraiba teve', len(acidentes_pb_2019), ' de acidentes com pessoas gravemente feridas em 2019')
print('A Paraiba teve', len(acidentes_pb_2020), ' de acidentes com pessoas gravemente feridas em 2020')
print('A Paraiba teve', len(acidentes_pb_2021), ' de acidentes com pessoas gravemente feridas em 2021')
print('Juntando todos os anos a Paraiba teve', len(acidentes_pb_3anos),
      ' de acidentes com pessoas gravemente feridas em 2021')


A Paraiba teve 367  de acidentes com pessoas gravemente feridas em 2018
A Paraiba teve 414  de acidentes com pessoas gravemente feridas em 2019
A Paraiba teve 360  de acidentes com pessoas gravemente feridas em 2020
A Paraiba teve 375  de acidentes com pessoas gravemente feridas em 2021
Juntando todos os anos a Paraiba teve 1516  de acidentes com pessoas gravemente feridas em 2021


<div class="alert alert-block alert-info">
    3° Tarefa
<div>

<div class="alert alert-block alert-info" style="background-color:#b9ceac; color:#000; border-radius:15px;" >
    a)
<div>

In [46]:
# Usando a função do groupby do pandas filtrei os dados assim obtendo o rank e converti o arquivo para csv.

rank = juntando_dados.groupby(['uf']).size().reset_index(name ='rank')
ranking = rank.sort_values(by ='rank', ascending = False )
ranking.to_csv("rank_de_todas_cidade_4anos.csv")
ranking

uf   rank
10  MG  34483
23  SC  32034
17  PR  30185
18  RJ  17982
22  RS  17754
25  SP  17236
4   BA  13793
8   GO  13233
15  PE  10566
7   ES  10543
12  MT   9101
5   CE   6505
11  MS   6147
20  RO   6067
14  PB   5993
19  RN   5527
16  PI   5057
9   MA   4631
6   DF   3978
13  PA   3884
1   AL   2528
24  SE   2255
26  TO   2205
0   AC   1044
21  RR    991
3   AP    626
2   AM    456

<div class="alert alert-block alert-info" style="background-color:#b9ceac; color:#000; border-radius:15px;" >
    b)
<div>

In [47]:
# usando a função do groupby do pandas filtrei os dados assim obtendo o rank e converti o arquivo para csv.

dia = juntando_dados.groupby(['dia_semana']).size().reset_index(name ='por_dia')
semana = dia.sort_values(by ='por_dia', ascending = False )
semana.to_csv("quantos_acidente_por_dia.csv")
semana

dia_semana  por_dia
5         sábado    43872
0        domingo    43816
4    sexta-feira    40945
3  segunda-feira    35712
2   quinta-feira    34471
1   quarta-feira    33347
6    terça-feira    32641

<div class="alert alert-block alert-info">
    4° Tarefa
<div>

<div class="alert alert-block alert-info" style="background-color:#b9ceac; color:#000; border-radius:15px;" >
    a)
<div>
   

In [49]:
#qual a causa mais frequente e a mais rara de acidentes registrados nos quatro anos
# a maior causa de acidentes usando groupby  
causa = juntando_dados.groupby(['causa_acidente']).size().reset_index(name ='numero_acidentes')
ranking_causa = causa.sort_values(by ='numero_acidentes', ascending = False )
ranking_causa.head(1)

causa_acidente  numero_acidentes
37  Falta de Atenção à Condução             73231

In [50]:
# A causa de menos acidentes nos 4 anos 
ranking_causa.tail(1)

causa_acidente  numero_acidentes
76  Sinalização encoberta                 3

<div class="alert alert-block alert-info" style="background-color:#b9ceac; color:#000; border-radius:15px;" >
    b)
<div>

In [51]:
# separando o ano do mes e do dia

mes18 = dados_prf_2018[['ano', 'mes', 'dia']] = dados_prf_2018['data_inversa'].str.split('-', expand=True)
mes18.iloc[:,1:2].rename(columns = {1:'mes'})

# pegando colunas que interressão para a analise

mes_2018b = dados_prf_2018.loc[:,['mes','pessoas','ilesos','feridos_graves']]
ano2018 = mes_2018b.groupby(by='mes').sum()
probalidade_ilesos_18 = (ano2018['ilesos']/ano2018['pessoas'])
probalidade_feridos_18 = (ano2018['feridos_graves']/ano2018['pessoas'])
probalidade_2018 = pd.DataFrame(probalidade_ilesos_18).rename(columns={0:'proporção ilesos em 2018'}).round(4)
probalidade_2018['proporção feridos graves em 2018'] = pd.DataFrame(probalidade_feridos_18).round(4)
probalidade_2018



proporção ilesos em 2018  proporção feridos graves em 2018
mes                                                            
01                     0.5133                            0.0906
02                     0.4983                            0.0922
03                     0.4815                            0.1019
04                     0.4302                            0.1128
05                     0.4222                            0.1145
06                     0.4251                            0.1164
07                     0.4375                            0.1103
08                     0.4271                            0.1064
09                     0.4207                            0.1155
10                     0.4302                            0.1126
11                     0.4334                            0.1154
12                     0.4329                            0.1081

In [52]:
# separando o ano do mes e do dia

mes19 = dados_prf_2019[['ano', 'mes', 'dia']] = dados_prf_2019['data_inversa'].str.split('-', expand=True)
mes19.iloc[:,1:2].rename(columns = {1:'mes'})

# pegando colunas que interressão para a analise

mes_2019b = dados_prf_2019.loc[:,['mes','pessoas','ilesos','feridos_graves']]
ano2019 = mes_2019b.groupby(by='mes').sum()

probalidade_ilesos_19 = (ano2019['ilesos']/ano2019['pessoas'])
probalidade_feridos_19 = (ano2019['feridos_graves']/ano2019['pessoas'])
probalidade_2019 = pd.DataFrame(probalidade_ilesos_19).rename(columns={0:'proporção ilesos em 2019'}).round(4)
probalidade_2019['proporção feridos graves em 2019'] = pd.DataFrame(probalidade_feridos_19).round(4)
probalidade_2019


proporção ilesos em 2019  proporção feridos graves em 2019
mes                                                            
01                     0.4242                            0.1133
02                     0.4241                            0.1068
03                     0.4332                            0.1146
04                     0.4157                            0.1060
05                     0.4152                            0.1208
06                     0.4238                            0.1216
07                     0.4368                            0.1167
08                     0.4131                            0.1167
09                     0.4211                            0.1107
10                     0.4189                            0.1198
11                     0.4258                            0.1110
12                     0.4223                            0.1144

In [53]:
# separando o ano do mes e do dia

mes20 = dados_prf_2020[['ano', 'mes', 'dia']] = dados_prf_2020['data_inversa'].str.split('-', expand=True)
mes20.iloc[:,1:2].rename(columns = {1:'mes'})

# pegando colunas que interressão para a analise

mes_2020b = dados_prf_2020.loc[:,['mes','pessoas','ilesos','feridos_graves']]
ano2020 = mes_2020b.groupby(by='mes').sum()
probalidade_ilesos_20 = (ano2020['ilesos']/ano2020['pessoas'])
probalidade_feridos_20 = (ano2020['feridos_graves']/ano2020['pessoas'])
probalidade_2020 = pd.DataFrame(probalidade_ilesos_20).rename(columns={0:'proporção ilesos em 2020'}).round(4)
probalidade_2020['proporção feridos graves em 2020'] = pd.DataFrame(probalidade_feridos_20).round(4)
probalidade_2020


proporção ilesos em 2020  proporção feridos graves em 2020
mes                                                            
01                     0.4156                            0.1075
02                     0.4200                            0.1114
03                     0.4010                            0.1126
04                     0.3763                            0.1179
05                     0.4013                            0.1159
06                     0.4233                            0.1159
07                     0.4243                            0.1086
08                     0.4234                            0.1192
09                     0.4160                            0.1174
10                     0.4110                            0.1197
11                     0.4070                            0.1204
12                     0.4123                            0.1220

In [54]:
# separando o ano do mes e do dia

mes21 = dados_prf_2021[['ano', 'mes', 'dia']] = dados_prf_2021['data_inversa'].str.split('-', expand=True)
mes21.iloc[:,1:2].rename(columns = {1:'mes'})

# pegando colunas que interressão para a analise
mes_2021b = dados_prf_2021.loc[:,['mes','pessoas','ilesos','feridos_graves']]
ano2021 = mes_2021b.groupby(by='mes').sum()

probalidade_ilesos_21 = (ano2021['ilesos']/ano2021['pessoas'])
probalidade_feridos_21 = (ano2021['feridos_graves']/ano2021['pessoas'])

probalidade_2021 = pd.DataFrame(probalidade_ilesos_21).rename(columns={0:'proporção ilesos em 2021'}).round(4)
probalidade_2021['proporção feridos graves em 2021'] = pd.DataFrame(probalidade_feridos_21).round(4)
probalidade_2021

proporção ilesos em 2021  proporção feridos graves em 2021
mes                                                            
01                     0.4257                            0.1124
02                     0.4135                            0.1157
03                     0.4109                            0.1146
04                     0.4061                            0.1226
05                     0.4158                            0.1250
06                     0.4307                            0.1093
07                     0.4319                            0.1134
08                     0.4212                            0.1193
09                     0.4170                            0.1218
10                     0.4145                            0.1176
11                     0.4268                            0.1209
12                     0.4212                            0.1121

<div class="alert alert-block alert-info" style="background-color:#b9ceac; color:#000; border-radius:15px;" >
    c)
<div>

In [55]:
# manipulação de dados para analise acidente durantes e antes da pandemia e porcentagens
antes_da_pandemia = len(pd.concat([dados_prf_2018,dados_prf_2019]))
durante_pandemia = len(pd.concat([dados_prf_2020,dados_prf_2021]))
antes = antes_da_pandemia /2
durante = durante_pandemia /2
diferença = antes - durante

# porcentagem de aumento ou diminição de casos referente ao ano anterior
porcentagem18_19 = (len(dados_prf_2019)-len(dados_prf_2018))/100
porcentagem19_20 = (len(dados_prf_2020)-len(dados_prf_2019))/100
porcentagem20_21 = (len(dados_prf_2021)-len(dados_prf_2020))/100


In [56]:
analise_pandemia = pd.DataFrame([antes,durante],
                                index=['Antes da Pandemia','Durante a Pandemia']
                               ).rename(columns={0:'Media de Incidentes'})
analise_pandemia


Media de Incidentes
Antes da Pandemia               68370.5
Durante a Pandemia              64031.5

In [57]:
defere = pd.DataFrame([diferença],index=['Diferença']).rename(columns={0:"Diferença entre Periodos de anos"})
defere


Diferença entre Periodos de anos
Diferença                            4339.0

Com essa analse da diferença de acidentes de antes da pandemia para durante vê que se tem uma Diferença 4339.0 a menos de acidentes 

In [60]:
porcentagem = pd.DataFrame([porcentagem18_19,porcentagem19_20 ,porcentagem20_21],
                           index=['Porcentagem de 2019 referente 2018','Porcentagem de 2020 referente 2019',
                                 'Porcentagem de 2021 referente 2020']).rename(columns={0:'Porcentagem %'})
porcentagem


Porcentagem %
Porcentagem de 2019 referente 2018         -18.49
Porcentagem de 2020 referente 2019         -38.98
Porcentagem de 2021 referente 2020           9.67

Analisando friamente os dados da porcentagem ver que nos de 2018 para 2019 já foi se vendo uma diminução nos numeros de acidentes que com a chegada pandema ouve uma diminuição significativa e em 2021 ouve um aumento de **9.67** em relação 2020